src/backend/gpopt/translate/CTranslatorQueryToDXL.cpp

```diff
@@ -684,6 +686,34 @@ CTranslatorQueryToDXL::TranslateQueryToDXL()
 	}
 }
 
+//---------------------------------------------------------------------------
+//	@function:
+//		CTranslatorQueryToDXL::GetTableDescr
+//
+//	@doc:
+//		Fills table descriptor of query. Assume, that subquery and further
+// 		paln would have only one table descriptor, which referes to target
+//		(result) relation, so it's worth to add  only query_id at table descr
+//		(without storing id of relation of pg query), for further
+//		deduplication target (result) relations.
```

Comment:
The second point about accounting of nested subqueries should look more clear. Smth like: "We uses queryid field to tag relations assigned to target one. In case of possible nested DML subqueries this field points to target relation of corresponding Query structure of subquery"